ChatApp/ChatApp.API/Startup.cs

```diff
@@ -28,16 +32,34 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-
             services.AddSingleton<IMongoClient>(s => new MongoClient(Configuration.GetSection("ChatDatabaseSettings:ConnectionString").Value));
-            services.AddScoped(s => new ChatDbContext(s.GetRequiredService<IMongoClient>(), Configuration.GetSection("ChatDatabaseSettings:DatabaseName").Value));
+            services.AddSingleton(s => new ChatDbContext(s.GetRequiredService<IMongoClient>(), Configuration.GetSection("ChatDatabaseSettings:DatabaseName").Value));
 
             services.AddScoped<IUserRepository, UserRepository>();
             services.AddScoped<IChatRepository, ChatRepository>();
             services.AddScoped<IUnitOfWork, UnitOfWork>();
 
             services.AddControllers();
 
+            services.AddAuthentication(options =>
+                {
+                    options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                    options.DefaultChallengeScheme = GoogleDefaults.AuthenticationScheme;
+                    options.DefaultSignOutScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                })
+               .AddCookie(options =>
+                   {
+                       options.LoginPath = new PathString("/Account/google-login");
+                       options.Cookie.HttpOnly = true;
+                   }
+               )
+               .AddGoogle(options =>
+               {
+                   options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                   options.ClientId = "158339939419-r3gh8lgsd3dp7nl1kihedga6q44080gk.apps.googleusercontent.com";
```

Comment:
It's better to move ClientId and ClientSecret to json config file.

Reply:
Done!